bench/wtperf/config.c

```diff
@@ -94,20 +94,29 @@ config_unescape(char *orig)
 }
 
 /*
- * config_assign --
- *	Assign the src config to the dest, any storage allocated in dest is
- * freed as a result.
+ * config_init --
+ *	CONFIG structure initializaiton, based on a source configuration.
  */
 int
-config_assign(CONFIG *dest, const CONFIG *src)
+config_init(CONFIG *dest, const CONFIG *src)
 {
 	CONFIG_QUEUE_ENTRY *conf_line, *tmp_line;
 	size_t i;
 	char *newstr, **pstr;
 
-	config_free(dest);
 	memcpy(dest, src, sizeof(CONFIG));
 
+	if (src->home != NULL)
+		dest->home = dstrdup(src->home);
+	if (src->monitor_dir != NULL)
+		dest->monitor_dir = dstrdup(src->monitor_dir);
+	if (src->partial_config != NULL)
+		dest->partial_config = dstrdup(src->partial_config);
+	if (src->reopen_config != NULL)
+		dest->reopen_config = dstrdup(src->reopen_config);
+	if (src->base_uri != NULL)
+		dest->base_uri = dstrdup(src->base_uri);
+
```

Comment:
Should "char *async_config" also be duplicated from destination to source config.
If free async_config in the config_free() function, wouldn't we be double freeing this one in case not NULL.